### ISSUE TITLE
Potential overflow when handling dates

### ISSUE THREAD

**Issue by bergercookie** [GothenburgBitFactory/taskwarrior#2432]
I'm running the following command which works fine:

```sh
$ task add description:"description1" entry:20250204T190645Z
#                                           ----
```

However setting the entry to 2065 instead of 2025 results in the following error

```sh
$ task add description:"description1" entry:'20650204T190645Z'
'-1293967291' is not a valid date in the '' format.
```

## Ouptut of `task diag`

```sh

task 2.5.3
   Platform: Linux

Compiler
    Version: 7.5.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
      Built: Mar 13 2021 20:48:56
      CMake: 3.20.0-rc4
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.5.18
 Build type: Release

Configuration
       File: /home/berger/.taskrc (found), 2347 bytes, mode 100664
       Data: /home/berger/.task (found), dir, mode 40775
    Locking: Enabled
         GC: Enabled
    $EDITOR: nvim
     Server: taskwarrior.inthe.am:53589
         CA: ~/.task/intheam.ca.cert.pem, readable, 2405 bytes
Certificate: ~/.task/intheam.certificate.pem, readable, 2710 bytes
        Key: ~/.task/intheam.key.pem, readable, 17747 bytes
      Trust: ignore hostname
    Ciphers: NORMAL
      Creds: inthe_am/nikos.koukis.utility/************************************

Hooks
     System: Enabled
   Location: /home/berger/.task/hooks
     Active: 
   Inactive: totals.py (not executable)unrecognized hook name

Tests
   Terminal: 120x30
       Dups: Scanned 4056 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 4056 tasks for broken references:
             No broken references found
```

**Comment by kwoot** [comment]
I assume you are aware of the current max date of 2038 in the unix world?
But it could also be you actually want to store a todo to remind yourserlf in 40 years.
My suggestion would be to concentrate your efforts on fixing the 2038 unix date problem.
After that, TW will easily follow in line.

**Comment by tbabej** [comment]
Hi @bergercookie, thanks for bringing this to our attention.

The 2.5.3 release unfortunately only still supports 32-bit timestamps, regardless of the underlying `time_t` size. This was fixed on the current development branch, `2.6.0`, see #2373 or the [current development branch release notes](https://github.com/GothenburgBitFactory/taskwarrior/blob/2.6.0/NEWS).

**Comment by bergercookie** [comment]
Thanks @tbabej, good to know that you 've already dealt with it!
